packages/edit-site/src/hooks/push-changes-to-global-styles/index.js

```diff
@@ -212,10 +213,14 @@ function PushChangesToGlobalStylesControl( {
 const withPushChangesToGlobalStyles = createHigherOrderComponent(
 	( BlockEdit ) => ( props ) => {
 		const blockEditingMode = useBlockEditingMode();
+		const supportsStyles =
+			hasBlockSupport( props.name, 'color' ) ||
+			hasBlockSupport( props.name, 'typography' );
+
```

Comment:
Are we absolutely 100% sure we only want to check for these two, not border and spacing?
Related: https://github.com/WordPress/gutenberg/pull/51621

Reply:
We can start with these two and widen the check as needed in the future.

Reply:
I'm a bit late to this party, sorry, but border etc should really be checked.

I was looking into this feature in https://github.com/WordPress/gutenberg/pull/51621 before being pulled onto higher priority tasks. Perhaps that could be of some help, or we roll whatever is need here into that PR?

Reply:
Thanks for the additional context, @aaronrobertshaw.

As I can see, the border will only be supported after #51621. I think it makes sense to adjust the checks there.

Reply:
Perhaps we need to wait with the border, until the border is fixed.
But it means the image blocks that currently have this "apply globally" option, won't,  until maybe 6.4.

Reply:
> But it means the image blocks that currently have this "apply globally" option, won't, until maybe 6.4.

It doesn't do anything for the Image block since the border still needs to be supported. Having an option that does nothing is more confusing.

Reply:
Thanks for taking a look.

I guess I wasn't comfortable half-fixing the feature. Particularly, when for spacing at least it's only a minor tweak to include it now, reducing the scope of what still remains broken. Combining the two fixes it all.

Reply:
> Apply this block’s typography, spacing, dimensions, and color styles to all Post Featured Image blocks.

This paragraph is misleading. If you check, the "style paths" feature only supports `color` and `typography`.

Unless I'm misunderstanding something, this PR can ship separately, and future PRs can adjust checks. We can also use a constant to keep track of `SUPPORTED_STYLES`.

Reply:
The comments above those style path constants note they are supposed to be duplicates from the block-editor and they are out of sync and incorrect. Hence missing things like spacing etc.

Reply:
Would something like this work - [512aeb4](https://github.com/WordPress/gutenberg/pull/52160/commits/512aeb4dcadac87d75008abff8846238536eabf3)?

Reply:
@aaronrobertshaw, any blockers to merge this PR?

Reply:
No, let's go with this. I've been caught up today on last-minute patterns updates and haven't had the time to revisit the other PR.